crates/primitives/starknet/src/transaction/mod.rs

```diff
@@ -94,7 +95,56 @@ impl Transaction {
             sender_address: StarknetContractAddress::try_from(StarkFelt::new(self.sender_address).unwrap()).unwrap(),
             calldata: self.call_entrypoint.to_starknet_call_entry_point().calldata,
             entry_point_selector: None,
-        })
+        }
+    }
+
+    /// Converts a transaction to a blockifier declare transaction
+    ///
+    /// # Arguments
+    ///
+    /// * `self` - The transaction to convert
+    ///
+    /// # Returns
+    ///
+    /// * `AccountTransaction` - The converted transaction
+    pub fn to_declare_tx(&self) -> DeclareTransaction {
+        DeclareTransaction {
+            transaction_hash: TransactionHash(StarkFelt::new(self.hash.0).unwrap()),
+            max_fee: Fee(2),
+            version: TransactionVersion(StarkFelt::new(self.version.into()).unwrap()),
+            signature: TransactionSignature(
+                self.signature.clone().into_inner().iter().map(|x| StarkFelt::new(x.0).unwrap()).collect(),
+            ),
+            nonce: Nonce(StarkFelt::new(self.nonce.into()).unwrap()),
+            sender_address: StarknetContractAddress::try_from(StarkFelt::new(self.sender_address).unwrap()).unwrap(),
+            class_hash: self.call_entrypoint.to_starknet_call_entry_point().class_hash.unwrap(),
+        }
+    }
+
+    /// Converts a transaction to a blockifier deploy account transaction
+    ///
+    /// # Arguments
+    ///
+    /// * `self` - The transaction to convert
+    ///
+    /// # Returns
+    ///
+    /// * `AccountTransaction` - The converted transaction
+    pub fn to_deploy_account_tx(&self) -> DeployAccountTransaction {
```

Comment:
```suggestion
    pub fn try_into_deploy_account_tx(&self) -> DeployAccountTransaction {
```
remove unwraps and return result